### DISCUSSION TITLE
Displaying large amounts of markers as performant as possible

### DISCUSSION THREAD

**Discussion by ccev** [maplibre/maplibre-gl-js#5207] [Q&A] [answered]
Hi. I'm trying to display as many markers as possible before having to cluster them. There is virtually no limit to how many I want to display at once.

The markers have to be an image and be clickable. Optionally, it would be nice to add custom styling to them.

Possible options I found were to add images via the symbol layer or using deck.gl to render the markers. Are there better ways to tackle this problem?

Thanks!

**Comment by neodescis** [comment, marked as answer]
Is there some reason you must use markers instead of a symbol layer? So far everything you've said is very easy to do with a symbol layer, which will support orders of magnitude more items than markers.

At the end of the day, markers are DOM elements, and there are only so many you can add before a browser starts to struggle.

**Comment by HarelM** [comment]
Yup, I agree, symbol layer sounds like what you would probably need.
